controller/src/beerocks/bml/bml.cpp

```diff
@@ -709,3 +709,60 @@ int bml_start_dcs_single_scan(BML_CTX ctx, const char *radio_mac, int dwell_time
     return pBML->start_dcs_single_scan(tlvf::mac_from_string(std::string(radio_mac)), dwell_time,
                                        channel_pool, channel_pool_size);
 }
+
+int bml_client_get_client_list(BML_CTX ctx, char *client_list, unsigned int *client_list_size)
+{
+    // Validate input parameters
+    if (!ctx && !client_list && !client_list_size) {
+        return (-BML_RET_INVALID_ARGS);
+    }
+
+    auto pBML = static_cast<bml_internal *>(ctx);
+    std::string temp_client_list;
+    int ret;
+    if (!(ret = pBML->client_get_client_list(temp_client_list, client_list_size))) {
+        mapf::utils::copy_string((char *)(client_list), temp_client_list.c_str(),
+                                 *client_list_size);
+        return BML_RET_OK;
+    }
+
+    // Clear the outgoing arguments
+    *client_list_size = 0;
+    *client_list      = 0;
```

Comment:
This is a "get" function, so why do you "delete" the things you got?
Also if it is a string you should put '\0' and not 0.

Reply:
This is a failed scenario, the "returning" values are incorrect